packages/preact/test/hooks/useSwatches.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { useSwatches } from "@preact/hooks/useSwatches"
+import { describe, expect, it } from "vitest"
+
+import { renderHookWithProviders } from "../mocks/renderHookWithProviders"
+
+const testSKUs = [
```

Comment:
try to have 3 * 3 skus and also make sure that all the intersection works correctly
[Red, Blue, Green] * [L, M, S]